bioio/plugins.py

```diff
@@ -40,13 +47,73 @@
 class PluginEntry(NamedTuple):
     entrypoint: EntryPoint
     metadata: ReaderMetadata
-    timestamp: float
 
 
 # global cache of plugins
 plugins_by_ext_cache: OrderedDict[str, List[PluginEntry]] = OrderedDict()
 
 
+def _normalize_extensions(exts: Sequence[str]) -> List[str]:
+    """
+    Normalize and deduplicate a sequence of extensions.
+
+    - Lowercases all entries
+    - Ensures each extension starts with '.'
+    - Removes duplicates while preserving first-seen order
+    """
+    normalized: List[str] = []
+    seen = set()
+
+    for ext in exts:
+        e = ext.lower()
+        if not e.startswith("."):
+            e = "." + e
+
+        if e not in seen:
+            seen.add(e)
+            normalized.append(e)
+
+    return normalized
+
+
+def _count_extension_families(exts: Sequence[str]) -> int:
```

Comment:
I feel like unit testing just this function would be very useful; actually, unit testing the results showing in the "Example" below is what I'd be curious about. Testing the counts is a bit simpler, but actually testing that it finds the expected families might be a more interesting test.